source/amx/amxaux.c

```diff
@@ -46,21 +48,24 @@ size_t AMXAPI aux_ProgramSize(char *filename)
 int AMXAPI aux_LoadProgram(AMX *amx, char *filename, void *memblock)
 {
   FILE *fp;
+  size_t size;
   AMX_HEADER hdr;
   int result, didalloc;
 
   /* open the file, read and check the header */
   if ((fp = fopen(filename, "rb")) == NULL)
     return AMX_ERR_NOTFOUND;
-  if (fread(&hdr, sizeof hdr, 1, fp) < 1)
-    return AMX_ERR_INIT;
-  amx_Align16(&hdr.magic);
-  amx_Align32((uint32_t *)&hdr.size);
-  amx_Align32((uint32_t *)&hdr.stp);
-  if (hdr.magic != AMX_MAGIC) {
+  size = fread(&hdr, sizeof hdr, 1, fp);
+  if (size < 1) {
+err_format:
```

Comment:
Can we have the goto here instead of below so that it jumps forward? or remove it at all